src/dgemm_kernel.rs

```diff
@@ -102,6 +110,442 @@ pub unsafe fn kernel_target_sse2(k: usize, alpha: T, a: *const T, b: *const T,
     kernel_fallback_impl(k, alpha, a, b, beta, c, rsc, csc)
 }
 
+#[inline(always)]
+#[cfg(any(target_arch="x86", target_arch="x86_64"))]
+unsafe fn kernel_x86_avx(k: usize, alpha: T, a: *const T, b: *const T,
+                         beta: T, c: *mut T, rsc: isize, csc: isize)
+{
+    debug_assert_ne!(k, 0);
+
+    let mut ab = [_mm256_setzero_pd(); MR];
+
+    // this kernel can operate in either transposition (C = A B or C^T = B^T A^T)
+    let prefer_row_major_c = rsc != 1;
+
+    // let (mut a, mut b) = if prefer_row_major_c { (a, b) } else { (b, a) };
+    // let (rsc, csc) = if prefer_row_major_c { (rsc, csc) } else { (csc, rsc) };
+    let (mut a, mut b) = (a, b);
+
+    // With MR=8, we load sets of 4 doubles from a
+    let mut a_0123 = _mm256_loadu_pd(a);
+    let mut a_4567 = _mm256_loadu_pd(a.add(4));
+
+    // With NR=4, we load 4 doubles from b
+    let mut b_0123 = _mm256_loadu_pd(b);
+
+    unroll_by_with_last!(4 => k, is_last, {
```

Comment:
4 is probably good, but it's something you can experiment with, if unrolling by 1,2,3,4,5,6,7,8 is better.